cli/ops/runtime_compiler.rs

```diff
@@ -71,16 +75,52 @@ struct TranspileArgs {
   options: Option<String>,
 }
 
+#[derive(Debug, Serialize)]
+struct RuntimeTranspileEmit {
+  source: String,
+  map: Option<String>,
+}
+
 async fn op_transpile(
   state: Rc<RefCell<OpState>>,
   args: Value,
   _data: BufVec,
 ) -> Result<Value, AnyError> {
   super::check_unstable2(&state, "Deno.transpile");
   let args: TranspileArgs = serde_json::from_value(args)?;
-  let cli_state = super::global_state2(&state);
-  let program_state = cli_state.clone();
-  let result =
-    runtime_transpile(program_state, &args.sources, &args.options).await?;
+
+  let mut compiler_options = tsc_config::TsConfig::new(json!({
+    "checkJs": true,
+    "emitDecoratorMetadata": false,
+    "jsx": "react",
+    "jsxFactory": "React.createElement",
+    "jsxFragmentFactory": "React.Fragment",
+    "inlineSourceMap": false,
+  }));
+
+  let user_options = if let Some(options) = args.options {
+    tsc_config::parse_config_with_ignored_options(&options)?
+  } else {
+    json!({})
+  };
+  compiler_options.merge(&user_options);
```

Comment:
Maybe...

```suggestion
  if let Some(options) = args.options {
    let user_options = tsc_config::parse_config_with_ignored_options(&options)?;
    compiler_options.merge(&user_options);
  }
```

Reply:
Actually why aren't you using `TsConfig::merge_user_config()`?  I wanted to get rid of `parse_config` when we dropped `tsc` anyways, as why can't these just live as methods on `TsConfig`?

Reply:
Not get rid of `parse_config` but make it non-public.